platform/android/vulkan/vulkan_context_android.cpp

```diff
@@ -31,14 +31,25 @@
 #include "vulkan_context_android.h"
 #include <vulkan/vulkan_android.h>
 
-#define VMA_IMPLEMENTATION
+#ifdef __GNUC__
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wimplicit-fallthrough"
+#pragma GCC diagnostic ignored "-Wmissing-field-initializers"
+#endif
```

Comment:
I don't think that's the right approach, we shouldn't have to disable specific warnings here, we want to ignore *all* warnings in thirdparty code as it's outside our jurisdiction. Otherwise when new warning types appear in vma, we'd have to blacklist each of them here. All thirdparty code in Godot should be built without warnings.

Instead, it should be done as in `drivers/vulkan/SCsub`:
```
env_thirdparty = env.Clone()
env_thirdparty.disable_warnings()
vma_sources = [thirdparty_dir + "/vk_mem_alloc.cpp"]
env_thirdparty.add_source_files(env.drivers_sources, vma_sources)
```

And here you'd just include `vk_mem_alloc.h`.

Reply:
Ok, that seems better indeed.

Would it make sense to have a separate `vulkan_android` dir in thirdparty to make sure `vk_mem_alloc.h` is included from the NDK and not from `thirdparty/vulkan`?